File: backend/internal/schemas/transaction.go
package schemas

import (
	"time"

	"github.com/google/uuid"
)

type SearchTransaction struct {
	DateFrom time.Time `json:"date_from" form:"date_from"`
	DateTo   time.Time `json:"date_to" form:"date_to"`
	Take     int       `json:"take" binding:"required"`
	Offset   int       `json:"offset" binding:"required"`
}

type InsertTransaction struct {
	AccountID       uuid.UUID  `json:"account_id" binding:"required,uuid"`
	CategoryID      uuid.UUID  `json:"category_id" binding:"required,uuid"`
	Amount          float64    `json:"amount" binding:"required,gt=0"`
	Type            string     `json:"type" binding:"required,oneof=income expense"`
	Description     *string    `json:"description" binding:"omitempty"`
	TransactionDate *time.Time `json:"transaction_date" binding:"omitempty"`
}

File: frontend/validation/transaction.ts
import { z } from "zod";

export const CreateTransactionDTOSchema = z.object({
    account_id: z.string({
        error: "Required"
    }).uuid(),
    category_id: z.string({
        error: "Required"
    }),
    amount: z.number({
        error: "Required"
    }),
    type: z.enum(["expense", "income"], {
        error: "Required"
    }),
    description: z.string().optional(),
    transaction_date: z.string().datetime().optional()
});

export type CreateTransactionDTO = z.infer<typeof CreateTransactionDTOSchema>;